### PR TITLE
Issue #92. Fixed date time list generation. Release candidate 7

### PR DESCRIPTION
**Pull request by llotall**

### CODE REVIEW

File: src/swagger/OpenAPITypesGuard.ts [hunk 1]
```diff
@@ -56,18 +56,30 @@ export class OpenAPITypesGuard {
...
     public isBoolean(schema: SchemaType): schema is IOpenAPI3BooleanSchema {
         return (schema as IOpenAPI3BooleanSchema)?.type === 'boolean';
     }
 
     public isSimple(schema: SchemaType): schema is OpenAPI3SimpleSchema {
-        return this.isGuid(schema) || this.isNumber(schema) || this.isString(schema) || this.isDate(schema) || this.isBoolean(schema);
+        return this.isGuid(schema) || this.isNumber(schema) || this.isDate(schema) || this.isString(schema) || this.isBoolean(schema);
+    }
+
+    private isStringType(schema: SchemaType) {
```
Comment (aleshchev) [inline src/swagger/OpenAPITypesGuard.ts:74]: isString

File: src/swagger/OpenAPITypesGuard.ts [hunk 2]
```diff
@@ -56,18 +56,30 @@ export class OpenAPITypesGuard {
     }
 
     public isString(schema: SchemaType): schema is IOpenAPI3StringSchema {
-        return (schema as IOpenAPI3StringSchema)?.type === 'string';
+        return this.isStringType(schema) && !this.isDateFormat(schema) && !this.isGuidFormat(schema);
```
Comment (aleshchev) [inline src/swagger/OpenAPITypesGuard.ts:59]: it's very strange. Maybe 
(schema as IOpenAPI3StringSchema)?.type === 'string' && schema.format === undefined

File: src/swagger/OpenAPITypesGuard.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export class OpenAPITypesGuard {
     }
 
     public isGuid(schema: SchemaType): schema is IOpenAPI3GuidSchema {
-        return this.isString(schema) && (schema as IOpenAPI3GuidSchema)?.format === 'uuid';
+        return this.isStringType(schema) && this.isGuidFormat(schema);
```
Comment (aleshchev) [inline src/swagger/OpenAPITypesGuard.ts:28]: Please return inline check. This way is more harder  understanding

File: src/swagger/OpenAPITypesGuard.ts [hunk 4]
```diff
@@ -56,18 +56,30 @@ export class OpenAPITypesGuard {
     }
 
     public isString(schema: SchemaType): schema is IOpenAPI3StringSchema {
-        return (schema as IOpenAPI3StringSchema)?.type === 'string';
+        return this.isStringType(schema) && !this.isDateFormat(schema) && !this.isGuidFormat(schema);
     }
 
     public isDate(schema: SchemaType): schema is IOpenAPI3DateSchema {
-        return this.isString(schema) && (schema as IOpenAPI3DateSchema)?.format === 'date-time';
+        return this.isStringType(schema) && this.isDateFormat(schema);
```
Comment (aleshchev) [inline src/swagger/OpenAPITypesGuard.ts:63]: inline check

File: src/swagger/OpenAPITypesGuard.ts (hunk 2 above)
Comment (sevaru) [reply src/swagger/OpenAPITypesGuard.ts:59]: or

isString(schema) && isFormatUndefined(..

**Review by sevaru — APPROVED** [review]

**Review by aleshchev — APPROVED** [review]

**Pull request merged**